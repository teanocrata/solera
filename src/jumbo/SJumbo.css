@import url('https://fonts.googleapis.com/css?family=Gloria+Hallelujah');

.jumbotron {
  height: 100vh;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
}

.jumbotron div {
  height: 30%;
  display: flex;
  justify-content: space-between;
  align-items: center;
  flex-direction: column;
}

.jumbotron h1 {
  font-family: "Gloria Hallelujah", cursive;
  color: forestgreen;
  max-width: 85%;
  text-align: center;
  font-size: 120px;
}

@media (max-width: 768px) {
  .jumbotron h1 {
    font-size: 90px;
  }
}

@media (max-width: 425px) {
  .jumbotron h1 {
    font-size: 70px;
  }
}

@media (max-width: 320px) {
  .jumbotron h1 {
    font-size: 50px;
  }
}
