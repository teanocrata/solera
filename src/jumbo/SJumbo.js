import React, { Component } from "react";
import { Jumbotron, Button } from "react-bootstrap";
import "./SJumbo.css";

class SJumbo extends Component {
  render() {
    const style = {
      jumbo: {
        backgroundColor: "transparent",
        height: this.props.heightPercentage
      },
      button: {
        backgroundColor: "lightgreen",
        color: "forestgreen",
        border: "none",
        boxShadow:
          "0 8px 16px 0 rgba(0,0,0,0.2), 0 6px 20px 0 rgba(0,0,0,0.19)",
        margin: "25% 0 25% 0",
        textTransform: "uppercase",
        fontWeight: "bold",
        fontSize: "35px"
      }
    };
    return this.props.jumbo ? (
      <Jumbotron className="text-center" style={style.jumbo}>
        <div>
          <h1>{this.props.jumbo.title}</h1>
          {this.props.jumbo.callToAction ? (
            <p>
              <Button style={style.button} bsSize="large">
                {this.props.jumbo.callToAction}
              </Button>
            </p>
          ) : null}
        </div>
      </Jumbotron>
    ) : null;
  }
}

export default SJumbo;
