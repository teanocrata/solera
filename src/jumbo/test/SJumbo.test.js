import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import SJumbo from "../SJumbo";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<SJumbo />, div);
});

test("SJumbo renders correctly", () => {
  const jumbo = renderer.create(<SJumbo />).toJSON();
  expect(jumbo).toMatchSnapshot();
});

test("SJumbo renders correctly with jumbo data", () => {
  const jumboData = {
    title: "Visitas guiadas",
    callToAction: "Ven a conocenos"
  };
  const jumbo = renderer.create(<SJumbo jumbo={jumboData} />).toJSON();
  expect(jumbo).toMatchSnapshot();
});

test("SJumbo renders correctly with call to acction", () => {
  const jumboData = {
    callToAction: "Ven a conocenos"
  };
  const jumbo = renderer.create(<SJumbo jumbo={jumboData} />).toJSON();
  expect(jumbo).toMatchSnapshot();
});

test("SJumbo renders correctly with title", () => {
  const jumboData = {
    title: "Visitas guiadas"
  };
  const jumbo = renderer.create(<SJumbo jumbo={jumboData} />).toJSON();
  expect(jumbo).toMatchSnapshot();
});
