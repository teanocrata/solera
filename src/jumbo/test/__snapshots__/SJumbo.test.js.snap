// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SJumbo renders correctly 1`] = `null`;

exports[`SJumbo renders correctly with call to acction 1`] = `
<div
  className="text-center jumbotron"
  style={
    Object {
      "backgroundColor": "transparent",
      "height": undefined,
    }
  }
>
  <div>
    <h1 />
    <p>
      <button
        className="btn btn-lg btn-default"
        disabled={false}
        style={
          Object {
            "backgroundColor": "lightgreen",
            "border": "none",
            "boxShadow": "0 8px 16px 0 rgba(0,0,0,0.2), 0 6px 20px 0 rgba(0,0,0,0.19)",
            "color": "forestgreen",
            "fontSize": "35px",
            "fontWeight": "bold",
            "margin": "25% 0 25% 0",
            "textTransform": "uppercase",
          }
        }
        type="button"
      >
        Ven a conocenos
      </button>
    </p>
  </div>
</div>
`;

exports[`SJumbo renders correctly with jumbo data 1`] = `
<div
  className="text-center jumbotron"
  style={
    Object {
      "backgroundColor": "transparent",
      "height": undefined,
    }
  }
>
  <div>
    <h1>
      Visitas guiadas
    </h1>
    <p>
      <button
        className="btn btn-lg btn-default"
        disabled={false}
        style={
          Object {
            "backgroundColor": "lightgreen",
            "border": "none",
            "boxShadow": "0 8px 16px 0 rgba(0,0,0,0.2), 0 6px 20px 0 rgba(0,0,0,0.19)",
            "color": "forestgreen",
            "fontSize": "35px",
            "fontWeight": "bold",
            "margin": "25% 0 25% 0",
            "textTransform": "uppercase",
          }
        }
        type="button"
      >
        Ven a conocenos
      </button>
    </p>
  </div>
</div>
`;

exports[`SJumbo renders correctly with title 1`] = `
<div
  className="text-center jumbotron"
  style={
    Object {
      "backgroundColor": "transparent",
      "height": undefined,
    }
  }
>
  <div>
    <h1>
      Visitas guiadas
    </h1>
  </div>
</div>
`;
